### PR TITLE
refactor: Migrate App initialization from knockout to React

### PR DESCRIPTION
**Pull request by przemvs**
Migrated Knockout rendering application to React.

Migrated with App-Loader, rendering content moved to id="wire-main".

### CODE REVIEW

File: src/script/main/app.ts [hunk 1]
```diff
@@ -175,23 +164,20 @@ class App {
...
-    private readonly apiClient: APIClient,
-  ) {
+  constructor(private readonly core: Core, private readonly apiClient: APIClient) {
     this.apiClient.on(APIClient.TOPIC.ON_LOGOUT, () => this.logout(SIGN_OUT_REASON.SESSION_EXPIRED, false));
     this.logger = getLogger('App');
-    this.appContainer = appContainer;
 
     new WindowHandler();
 
+    // done.
```
Comment (atomrc) [inline src/script/main/app.ts]: ???
```suggestion
```

File: src/script/main/app.ts [hunk 2]
```diff
@@ -175,23 +164,20 @@ class App {
...
     this.service = this._setupServices();
+    // done.
```
Comment (atomrc) [inline src/script/main/app.ts]:
```suggestion
```

File: src/script/main/app.ts [hunk 3]
```diff
@@ -401,23 +400,21 @@ class App {
...
-      if (this.apiClient.backendFeatures.isFederated) {
+
+      if (this.apiClient.backendFeatures.isFederated && this.repository.storage.storageService.db) {
         // Migrate all existing session to fully qualified ids (if need be)
-        const sessions = this.repository.storage.storageService.db?.sessions;
-        const domain = this.apiClient.context?.domain;
-        if (sessions && domain) {
-          await migrateToQualifiedSessionIds(sessions, domain);
-        }
+        await migrateToQualifiedSessionIds(this.repository.storage.storageService.db.sessions, context.domain ?? '');
```
Comment (atomrc) [inline src/script/main/app.ts]: The condition on `domain` is important here. We should never migrate with an empty domain!!!

File: src/script/main/app.ts (hunk 1 above)
Comment (przemvs) [reply src/script/main/app.ts]: Nice catch, forgot to remove it

**Review by atomrc — CHANGES_REQUESTED** [review]

File: src/page/index.ejs [hunk 4]
```diff
@@ -24,18 +24,7 @@
...
-      data-bind="css: mainClasses()"
-      params="root: $root"
-      data-uie-name="status-webapp"
-      data-uie-value="is-loading"
-    ></app-container>
-
-    <div id="loading-screen" class="loading-screen">
-      <loading-bar params="message: loadingMessage, progress: loadingProgress"></loading-bar>
-    </div>
+    <div id="wire-app"></div>
```
Comment (atomrc) [inline src/page/index.ejs]: Isn't it a problem that we have one extra wrapping div now? Is everything's good with scrolling or styling?

File: src/page/index.ejs [hunk 5]
```diff
@@ -24,18 +24,7 @@
   </head>
 
   <body>
-    <app-container
-      id="wire-main"
-      class="off"
-      data-bind="css: mainClasses()"
-      params="root: $root"
-      data-uie-name="status-webapp"
-      data-uie-value="is-loading"
```
Comment (atomrc) [inline src/page/index.ejs:33]: Is this migrated? I don't think I have seen those in the react component?

File: src/page/index.ejs (hunk 4 above)
Comment (przemvs) [reply src/page/index.ejs]: I didn't spot any problems

File: src/page/index.ejs (hunk 5 above)
Comment (przemvs) [reply src/page/index.ejs:33]: Added it ! :)

File: src/script/main/app.ts [hunk 6]
```diff
@@ -491,13 +491,11 @@ class App {
       await conversationRepository.conversationLabelRepository.loadLabels();
 
       telemetry.timeStep(AppInitTimingsStep.APP_LOADED);
-      this._showInterface();
+      // this._showInterface();
```
Comment (PatrykBuniX) [inline src/script/main/app.ts]: what about this part?

File: src/script/page/AppMain.tsx [hunk 7]
```diff
@@ -129,12 +272,12 @@ const AppContainer: FC<AppContainerProps> = ({root}) => {
...
           <GroupCreationModal userState={userState} teamState={teamState} />
-        </main>
+        </>
       </RootProvider>
     </StyledApp>
   );
 };
 
-registerReactComponent('app-container', AppContainer);
+// registerReactComponent('app-container', AppMain);
```
Comment (PatrykBuniX) [inline src/script/page/AppMain.tsx]: to remove?

Comment (przemvs) [reply src/script/page/AppMain.tsx]: forgot :D

File: src/script/main/app.ts (hunk 6 above)
Comment (przemvs) [reply src/script/main/app.ts]: Forgot to remove :D

**Review by atomrc — APPROVED** [review]

**Pull request merged**